core/src/main/java/com/yandex/yoctodb/util/immutable/impl/VariableLengthByteArrayIndexedList.java

```diff
@@ -70,6 +72,54 @@ public Buffer get(final int i) {
         return elements.slice(start, end - start);
     }
 
+    @Override
+    public long getLongUnsafe(final int i) {
+        assert 0 <= i && i < elementCount;
+
+        final long base = ((long) i) << 3;
+        final long start = offsets.getLong(base);
+
+        return elements.getLong(start) ^ Long.MIN_VALUE;
+    }
+
+    @Override
+    public int getIntUnsafe(final int i) {
+        assert 0 <= i && i < elementCount;
+
+        final long base = ((long) i) << 3;
+        final long start = offsets.getLong(base);
+
+        return elements.getInt(start) ^ Integer.MIN_VALUE;
+    }
+
+    @Override
+    public short getShortUnsafe(final int i) {
+        assert 0 <= i && i < elementCount;
+
+        final long base = ((long) i) << 3;
+        final long start = offsets.getLong(base);
+        final long end = offsets.getLong(base + 8L);
+
+        assert end - start == Short.BYTES;
+        final int res = elements.getShort(start) ^ Short.MIN_VALUE;
+
+        return Shorts.checkedCast(res);
+    }
+
+    @Override
+    public char getCharUnsafe(final int i) {
+        assert 0 <= i && i < elementCount;
+
+        final long base = ((long) i) << 3;
+        final long start = offsets.getLong(base);
+        final long end = offsets.getLong(base + 8L);
+
+        assert end - start == Character.BYTES;
+        final int res = elements.getShort(start) ^ Character.MIN_VALUE;
```

Comment:
Danger! `Short` isn't the same as `Character`!
Also, redundant XOR with `Character.MIN_VALUE`

Reply:
Typo